### PR TITLE
Exploit the more complete DatasetDescriptor

### PR DESCRIPTION
**Pull request by forman**
UI changes wrt https://github.com/CCI-Tools/cate/pull/1006:

Datasets availablitiy and capabilities are now derived from obtained (ODP store) and computed (Zarr store) `verification_flags` (called `capabilities` in Cate App):

- "open"  - dataset can be opened
- "constrain_time" - dataset can be opened and time can be constrained
- "constrain_region" - dataset can be opened and region (bbox) can be constrained
- "write_zarr" - dataset can be opened and cached

### CODE REVIEW

**Comment by forman** [comment]
@dzelge FYI!

File: src/renderer/state-util.ts [hunk 1]
```diff
@@ -15,7 +20,7 @@ import {
     SavedLayers,
     SPLIT_MODE_OFF,
     TableViewDataState,
-    VariableDataRefState,
+    VariableDataRefState, VariableDescriptor,
```
Comment (TonioF) [inline src/renderer/state-util.ts:23]: break the line

File: src/renderer/state-util.ts [hunk 2]
```diff
@@ -101,35 +106,84 @@ export function getDataSourceUrls(dataSource: DataSourceState): DataSourceUrls {
...
+                            && isCoord1DInDataset(dsd, 'lat');
+    if (canConstrainTime && canConstrainRegion) {
+        return ["open", "constrain_time", "constrain_region"];
+    } else if (canConstrainTime) {
+        return ["open", "constrain_time"];
+    } else if (canConstrainRegion) {
+        return ["open", "constrain_region"];
+    }
+    if (isString(dsd.type_specifier)) {
+        // dataSource.typeSpecifier have been introduced for ESA CCI datasets only
```
Comment (TonioF) [inline src/renderer/state-util.ts:126]:
```suggestion
        // dataSource.typeSpecifier has been introduced for ESA CCI datasets only
```

**Review by AliceBalfanz — APPROVED** [review]
works well, thank you for implementing :)

**Review by TonioF — APPROVED** [review]

**Pull request merged**